fixing a problem with bins in choropleth legend

alvare opened the pull request:
When adding `legend=True` and `scheme` to `plot_dataframe` the values of the range where being miscalculated.

For example doing this:

```python
import geopandas as gpd
import matplotlib.pyplot as plt
import numpy as np

df = gpd.read_file(gpd.datasets.get_path('naturalearth_lowres'))
df['data'] = np.linspace(-10, 10, df.count()[0])
df.plot(column='data', cmap='RdBu_r', scheme='quantiles', legend=True)
plt.savefig('before.png')
```
Would result in:
![before](https://user-images.githubusercontent.com/2049010/27263036-da225b96-5438-11e7-9393-6bdc0dace9b9.png)

Notice the `0 - -6` range of the first color., which makes no sense.

With the fix it looks like this:
![after](https://user-images.githubusercontent.com/2049010/27263038-daf8d810-5438-11e7-95e9-1e7b00093bb8.png)

alvare commented:
I have no idea what `codecov/patch - 0% of diff hit` means. Do I need to change something?

jorisvandenbossche commented:
@alvare Thanks for the fix! 
Would you like to add a test for this?

(you can ignore the codecov)

alvare commented:
Sure! I just did. Also, I had broken the `TestPySALPlotting.test_legend` test which was incorrectly expecting the bins to start at 0, which was exactly the bug I solved, so I fixed that test and added one for negative values just in case: `TestPySALPlotting.test_negative_legend`.

Cheers!
